Nova/BellPerson/Solver.lean

```diff
@@ -0,0 +1,34 @@
+import YatimaStdLib.Bit
+
+structure DensityTracker where
+  bv : Array Bit
+  total_density : USize
+
+def new_tracker : DensityTracker := DensityTracker.mk #[] 0
+
+def add_element (tr : DensityTracker) : DensityTracker :=
+  let new_bv :=  Array.push tr.bv .zero
+  DensityTracker.mk new_bv tr.total_density
+
+structure SatisfyingAssignment (G : Type _) where
```

Comment:
Do we want to constrain `G` here to be of the elliptic curve's group type?

Reply:
Why don't we leave this type an arbitrary and declare that `G` is a group whenever we need that?